source/Smoldyn/smolsim.cpp

```diff
@@ -2464,19 +2464,20 @@ int simulatetimestep(simptr sim) {
 	er=filDynamics(sim);
 	if(er) return 11;
 
+#ifdef ENABLE_PYTHON_CALLBACK
+        for(unsigned int i = 0; i < sim->ncallbacks; i++) {
+            auto callback = sim->callbacks[i];
+            if((0 == (sim->simstep % callback->getStep())) && callback->isValid()) {
+                callback->evalAndUpdate(sim->time);
+            }
+        }
```

Comment:
Fixed a bug. Moving callback before the time is updated. In the callback, the time was always 1 dt ahead because of this.